### DISCUSSION TITLE
Firefly III v6.1.1 error：SQLSTATE[HY000] [2002] Connection refused

### DISCUSSION THREAD

**Discussion by realzsan3** [firefly-iii/firefly-iii#8316] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

错误类型为“Illuminate\Database\QueryException”

### Debug information

```
#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php(976): Illuminate\Database\Connection->runQueryCallback()
#1 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php(955): Illuminate\Database\Connection->tryAgainIfCausedByLostConnection()
#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php(778): Illuminate\Database\Connection->handleQueryException()
#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php(407): Illuminate\Database\Connection->run()
#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2751): Illuminate\Database\Connection->select()
#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2740): Illuminate\Database\Query\Builder->runSelect()
#6 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(3294): Illuminate\Database\Query\Builder->Illuminate\Database\Query\{closure}()
#7 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2739): Illuminate\Database\Query\Builder->onceWithColumns()
#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(737): Illuminate\Database\Query\Builder->get()
#9 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(721): Illuminate\Database\Eloquent\Builder->getModels()
#10 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/Relation.php(203): Illuminate\Database\Eloquent\Builder->get()
#11 /var/www/html/app/Helpers/Collector/GroupCollector.php(457): Illuminate\Database\Eloquent\Relations\Relation->get()
#12 /var/www/html/app/Helpers/Collector/GroupCollector.php(430): FireflyIII\Helpers\Collector\GroupCollector->getGroups()
#13 /var/www/html/app/Repositories/Category/OperationsRepository.php(62): FireflyIII\Helpers\Collector\GroupCollector->getExtractedJournals()
#14 /var/www/html/app/Http/Controllers/Chart/CategoryReportController.php(71): FireflyIII\Repositories\Category\OperationsRepository->listExpenses()
#15 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): FireflyIII\Http\Controllers\Chart\CategoryReportController->budgetExpense()
#16 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(43): Illuminate\Routing\Controller->callAction()
#17 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Route.php(259): Illuminate\Routing\ControllerDispatcher->dispatch()
#18 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Route.php(205): Illuminate\Routing\Route->runController()
#19 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(806): Illuminate\Routing\Route->run()
#20 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(144): Illuminate\Routing\Router->Illuminate\Routing\{closure}()
#21 /var/www/html/app/Http/Controllers/Chart/CategoryReportController.php(63): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#22 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(165): FireflyIII\Http\Controllers\Chart\CategoryReportController->FireflyIII\Http\Controllers\Chart\{closure}()
#23 /var/www/html/app/Http/Controllers/Controller.php(118): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#24 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(165): FireflyIII\Http\Controllers\Controller->FireflyIII\Http\Controllers\{closure}()
#25 /var/www/html/app/Http/Middleware/InterestingMessage.php(71): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#26 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): FireflyIII\Http\Middleware\InterestingMessage->handle()
#27 /var/www/html/app/Http/Middleware/Binder.php(75): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#28 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): FireflyIII\Http\Middleware\Binder->handle()
#29 /var/www/html/app/Http/Middleware/Range.php(56): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#30 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): FireflyIII\Http\Middleware\Range->handle()
#31 /var/www/html/vendor/jc5/google2fa-laravel/src/Middleware.php(29): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#32 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): PragmaRX\Google2FALaravel\Middleware->handle()
#33 /var/www/html/app/Http/Middleware/Authenticate.php(66): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#34 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): FireflyIII\Http\Middleware\Authenticate->handle()
#35 /var/www/html/vendor/laravel/passport/src/Http/Middleware/CreateFreshApiToken.php(63): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#36 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Laravel\Passport\Http\Middleware\CreateFreshApiToken->handle()
#37 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/Middleware/AuthenticateSession.php(60): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#38 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Session\Middleware\AuthenticateSession->handle()
#39 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(78): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#40 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\VerifyCsrfToken->handle()
#41 /var/www/html/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#42 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\View\Middleware\ShareErrorsFromSession->handle()
#43 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(121): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#44 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\Session\Middleware\StartSession->handleStatefulRequest()
#45 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Session\Middleware\StartSession->handle()
#46 /var/www/html/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#47 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle()
#48 /var/www/html/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#49 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Cookie\Middleware\EncryptCookies->handle()
#50 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(119): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#51 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(805): Illuminate\Pipeline\Pipeline->then()
#52 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(784): Illuminate\Routing\Router->runRouteWithinStack()
#53 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(748): Illuminate\Routing\Router->runRoute()
#54 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(737): Illuminate\Routing\Router->dispatchToRoute()
#55 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(200): Illuminate\Routing\Router->dispatch()
#56 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(144): Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}()
#57 /var/www/html/app/Http/Middleware/InstallationId.php(47): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#58 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): FireflyIII\Http\Middleware\InstallationId->handle()
#59 /var/www/html/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php(39): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#60 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Http\Middleware\TrustProxies->handle()
#61 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#62 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle()
#63 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\ConvertEmptyStringsToNull->handle()
#64 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#65 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle()
#66 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\TrimStrings->handle()
#67 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#68 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\ValidatePostSize->handle()
#69 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(99): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#70 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\PreventRequestsDuringMaintenance->handle()
#71 /var/www/html/app/Http/Middleware/SecureHeaders.php(48): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#72 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): FireflyIII\Http\Middleware\SecureHeaders->handle()
#73 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(119): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#74 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(175): Illuminate\Pipeline\Pipeline->then()
#75 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(144): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter()
#76 /var/www/html/public/index.php(73): Illuminate\Foundation\Http\Kernel->handle()
#77 {main}
```

### Expected behaviour

Firefly III v6.1.1 发生了错误：SQLSTATE[HY000] [2002] Connection refused (Connection: mysql, SQL: select `transaction_groups`.`id` as `transaction_group_id`, `transaction_groups`.`user_id` as `user_id`, `transaction_groups`.`user_group_id` as `user_group_id`, `transaction_groups`.`created_at` as `created_at`, `transaction_groups`.`updated_at` as `updated_at`, `transaction_groups`.`title` as `transaction_group_title`, `transaction_journals`.`id` as `transaction_journal_id`, `transaction_journals`.`transaction_type_id`, `transaction_journals`.`description`, `transaction_journals`.`date`, `transaction_journals`.`order`, `transaction_types`.`type` as `transaction_type_type`, `source`.`id` as `source_transaction_id`, `source`.`account_id` as `source_account_id`, `source`.`reconciled`, `source`.`amount` as `amount`, `source`.`transaction_currency_id` as `currency_id`, `currency`.`code` as `currency_code`, `currency`.`name` as `currency_name`, `currency`.`symbol` as `currency_symbol`, `currency`.`decimal_places` as `currency_decimal_places`, `source`.`foreign_amount` as `foreign_amount`, `source`.`foreign_currency_id` as `foreign_currency_id`, `foreign_currency`.`code` as `foreign_currency_code`, `foreign_currency`.`name` as `foreign_currency_name`, `foreign_currency`.`symbol` as `foreign_currency_symbol`, `foreign_currency`.`decimal_places` as `foreign_currency_decimal_places`, `destination`.`account_id` as `destination_account_id`, `categories`.`id` as `category_id`, `categories`.`name` as `category_name`, `source_account`.`name` as `source_account_name`, `source_account`.`iban` as `source_account_iban`, `source_account_type`.`type` as `source_account_type`, `dest_account`.`name` as `destination_account_name`, `dest_account`.`iban` as `destination_account_iban`, `dest_account_type`.`type` as `destination_account_type`, `budgets`.`id` as `budget_id`, `budgets`.`name` as `budget_name` from `transaction_journals` left join `transaction_groups` on `transaction_journals`.`transaction_group_id` = `transaction_groups`.`id` left join `transactions` as `source` on `source`.`transaction_journal_id` = `transaction_journals`.`id` and `source`.`amount` < 0 left join `transactions` as `destination` on `destination`.`transaction_journal_id` = `transaction_journals`.`id` and `destination`.`amount` > 0 left join `transaction_types` on `transaction_types`.`id` = `transaction_journals`.`transaction_type_id` left join `transaction_currencies` as `currency` on `currency`.`id` = `source`.`transaction_currency_id` left join `transaction_currencies` as `foreign_currency` on `foreign_currency`.`id` = `source`.`foreign_currency_id` left join `category_transaction_journal` on `category_transaction_journal`.`transaction_journal_id` = `transaction_journals`.`id` left join `categories` on `category_transaction_journal`.`category_id` = `categories`.`id` left join `accounts` as `source_account` on `source_account`.`id` = `source`.`account_id` left join `account_types` as `source_account_type` on `source_account_type`.`id` = `source_account`.`account_type_id` left join `accounts` as `dest_account` on `dest_account`.`id` = `destination`.`account_id` left join `account_types` as `dest_account_type` on `dest_account_type`.`id` = `dest_account`.`account_type_id` left join `budget_transaction_journal` on `budget_transaction_journal`.`transaction_journal_id` = `transaction_journals`.`id` left join `budgets` on `budget_transaction_journal`.`budget_id` = `budgets`.`id` where `transaction_journals`.`user_id` = 1 and `transaction_journals`.`user_id` is not null and `transaction_groups`.`deleted_at` is null and `transaction_journals`.`deleted_at` is null and `source`.`deleted_at` is null and `destination`.`deleted_at` is null and `transaction_journals`.`date` >= 2023-01-01 00:00:00 and `transaction_journals`.`date` <= 2023-12-31 23:59:59 and `transaction_types`.`type` in (Withdrawal) and (`source`.`account_id` in (28, 81, 1, 10, 14, 8, 3, 12, 6, 27, 26, 4, 31, 29, 16, 30) or `destination`.`account_id` in (28, 81, 1, 10, 14, 8, 3, 12, 6, 27, 26, 4, 31, 29, 16, 30)) and `categories`.`id` in (1, 2, 3, 4, 5, 6, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 26, 27, 28, 29) and `transaction_journals`.`deleted_at` is null order by `transaction_journals`.`date` desc, `transaction_journals`.`order` asc, `transaction_journals`.`id` desc, `transaction_journals`.`description` desc, `source`.`amount` desc)。

**Comment by iMuhammadMustafa** [comment]
After spending the whole day trying to debug it, I finally got it work with this docker compose. not sure what exactly made it finally work but this has been one of the most time consuming service to install 





```yml
version: '3.3'

services:
  app:
    image: fireflyiii/core
    hostname: app
    container_name: app
    restart: always
    volumes:
      - ./volumes/firefly_iii_upload:/var/www/html/storage/upload
    environment:
        - APP_KEY=heARL5nQXV1Vq4vcV9NPK2k31z5XR4JS
        - DB_HOST=db
        - DB_PORT=3306 
        - DB_CONNECTION=mysql 
        - DB_DATABASE=fireflydb
        - DB_USERNAME=fireflyuser
        - DB_PASSWORD=fireflypassword
        - APP_ENV=local
    ports:
      - 9113:8080
    depends_on:
      - db
    networks:
      - skynet
  db:
    image: mysql:8.2.0
    command: --default-authentication-plugin=mysql_native_password
    hostname: db
    container_name: db
    restart: always
    #env_file: .db.env
    environment:
        - MYSQL_ROOT_PASSWORD=fireflypassword
        - MYSQL_DATABASE=fireflydb        
        - MYSQL_USER=fireflyuser
        - MYSQL_PASSWORD=fireflypassword
    volumes:
      - ./volumes/firefly_iii_db:/var/lib/mysql
    networks:
      - skynet
networks:
  skynet:

  #cron:
    #
    # To make this work, set STATIC_CRON_TOKEN in your .env file or as an environment variable and replace REPLACEME below
    # The STATIC_CRON_TOKEN must be *exactly* 32 characters long
    #
   # image: alpine
   # restart: always
   # container_name: firefly_iii_cron
   # command: sh -c "echo \"0 3 * * * wget -qO- http://app:8080/api/v1/cron/REPLACEME\" | crontab - && crond -f -L /dev/stdout"
 ```

**Comment by rajviraujla22** [comment]
Hi, I have a similar issue. I have installed firefly-iii in docker on Synology nas. My importer page opens up but the firefly login page gives similar error. it says my user doesnot have access to the database. I could not figure out why is that. I will add my files.

Could not poll the database: SQLSTATE[HY000] [1045] Access denied for user 'firefly'@'192.168.64.4' (using password: YES) (Connection: mysql, SQL: select id, name, data from configuration where name = is_demo_site and configuration.deleted_at is null limit 1)

This error occured in file /var/www/html/app/Support/FireflyConfig.php on line 65 with code 0.

docker-compose.yml

version: '3.3'

services:
  app:
    image: fireflyiii/core:latest
    hostname: app
    container_name: firefly_iii_core
    networks:
      - firefly_iii
    restart: always
    volumes:
      - firefly_iii_upload:/var/www/html/storage/upload
    env_file: .env
    ports:
      - '901:8080'
    depends_on:
      - db
  db:
    image: mariadb:lts
    hostname: db
    container_name: firefly_iii_db
    networks:
      - firefly_iii
    restart: always
    env_file: .db.env
    volumes:
      - firefly_iii_db:/var/lib/mysql

  importer:
    image: fireflyiii/data-importer:latest
    hostname: importer
    restart: always
    container_name: firefly_iii_importer
    networks:
      - firefly_iii
    ports:
      - '902:8080'
    depends_on:
      - app
    env_file: .importer.env

  cron:
    
    image: alpine
    container_name: firefly_iii_cron
    restart: always
    command: sh -c "echo \"0 3 * * * wget -qO- http://app:8080/api/v1/cron/V6cu2koASQDoCBnY8fPY9cD7myEQrjUS\" | crontab - && crond -f -L /dev/stdout"
    networks:
      - firefly_iii
volumes:
   firefly_iii_upload:
   firefly_iii_db:

networks:
  firefly_iii:
    driver: bridge

.env


APP_ENV=production
APP_DEBUG=false

SITE_OWNER=mail@example.com
APP_KEY=V6cu2koASQDoCBnY8fPY9cD7myEQrjUS
DEFAULT_LANGUAGE=en_US

DEFAULT_LOCALE=equal

TZ=America/Edmonton

TRUSTED_PROXIES=
LOG_CHANNEL=stack


APP_LOG_LEVEL=notice



AUDIT_LOG_LEVEL=emergency
AUDIT_LOG_CHANNEL=
PAPERTRAIL_HOST=
PAPERTRAIL_PORT=

DB_CONNECTION=mysql
DB_HOST=db
DB_PORT=3306
DB_DATABASE=firefly
DB_USERNAME=firefly
DB_PASSWORD=myfirefly

DB_SOCKET=

MYSQL_USE_SSL=false
MYSQL_SSL_VERIFY_SERVER_CERT=true

MYSQL_SSL_CAPATH=/etc/ssl/certs/
MYSQL_SSL_CA=
MYSQL_SSL_CERT=
MYSQL_SSL_KEY=
MYSQL_SSL_CIPHER=

PGSQL_SSL_MODE=prefer
PGSQL_SSL_ROOT_CERT=null
PGSQL_SSL_CERT=null
PGSQL_SSL_KEY=null
PGSQL_SSL_CRL_FILE=null


PGSQL_SCHEMA=public

CACHE_DRIVER=file
SESSION_DRIVER=file


REDIS_SCHEME=tcp

REDIS_PATH=
REDIS_HOST=127.0.0.1
REDIS_PORT=6379

REDIS_USERNAME=
REDIS_PASSWORD=

REDIS_DB="0"
REDIS_CACHE_DB="1"

COOKIE_PATH="/"
COOKIE_DOMAIN=
COOKIE_SECURE=false
COOKIE_SAMESITE=lax

MAIL_MAILER=log
MAIL_HOST=null
MAIL_PORT=2525
MAIL_FROM=changeme@example.com
MAIL_USERNAME=null
MAIL_PASSWORD=null
MAIL_ENCRYPTION=null
MAIL_SENDMAIL_COMMAND=

MAILGUN_DOMAIN=
MAILGUN_SECRET=

MAILGUN_ENDPOINT=api.mailgun.net

MANDRILL_SECRET=
SPARKPOST_SECRET=

SEND_ERROR_MESSAGE=true

SEND_REPORT_JOURNALS=true

ENABLE_EXTERNAL_MAP=false


ENABLE_EXCHANGE_RATES=false


ENABLE_EXTERNAL_RATES=false

MAP_DEFAULT_LAT=51.983333
MAP_DEFAULT_LONG=5.916667
MAP_DEFAULT_ZOOM=6



VALID_URL_PROTOCOLS=




AUTHENTICATION_GUARD=web


AUTHENTICATION_GUARD_HEADER=REMOTE_USER
AUTHENTICATION_GUARD_EMAIL=


PASSPORT_PRIVATE_KEY=
PASSPORT_PUBLIC_KEY=


CUSTOM_LOGOUT_URL=


DISABLE_FRAME_HEADER=false


DISABLE_CSP_HEADER=false


TRACKER_SITE_ID=
TRACKER_URL=


ALLOW_WEBHOOKS=false


STATIC_CRON_TOKEN=V6cu2koASQDoCBnY8fPY9cD7myEQrjUS


DKR_BUILD_LOCALE=false

DKR_CHECK_SQLITE=true


DKR_RUN_MIGRATION=true


DKR_RUN_UPGRADE=true

DKR_RUN_VERIFY=true

DKR_RUN_REPORT=true

DKR_RUN_PASSPORT_INSTALL=true


APP_NAME=FireflyIII
BROADCAST_DRIVER=log
QUEUE_DRIVER=sync
CACHE_PREFIX=firefly
PUSHER_KEY=
IPINFO_TOKEN=
PUSHER_SECRET=
PUSHER_ID=
DEMO_USERNAME=
DEMO_PASSWORD=


FIREFLY_III_LAYOUT=v1


APP_URL=http://localhost

.db.env

MYSQL_RANDOM_ROOT_PASSWORD=yes
MYSQL_USER=firefly
MYSQL_PASSWORD=myfirefly
MYSQL_DATABASE=firefly

.importer.env

FIREFLY_III_URL=
VANITY_URL=
FIREFLY_III_ACCESS_TOKEN=
FIREFLY_III_CLIENT_ID=
NORDIGEN_ID=
NORDIGEN_KEY=
NORDIGEN_SANDBOX=false


SPECTRE_APP_ID=
SPECTRE_SECRET=


USE_CACHE=false

IGNORE_DUPLICATE_ERRORS=false
IGNORE_NOT_FOUND_TRANSACTIONS=false

AUTO_IMPORT_SECRET=


CAN_POST_AUTOIMPORT=false

CAN_POST_FILES=false


IMPORT_DIR_ALLOWLIST=

FALLBACK_IN_DIR=false




VERIFY_TLS_SECURITY=true


JSON_CONFIGURATION_DIR=


CONNECTION_TIMEOUT=31.41

APP_ENV=local
APP_DEBUG=false
LOG_CHANNEL=stack

LOG_RETURN_JSON=false

LOG_LEVEL=debug


TRUSTED_PROXIES=
TZ=America/Edmonton


ENABLE_MAIL_REPORT=false


EXPECT_SECURE_URL=false


MAIL_MAILER=


MAIL_DESTINATION=noreply@example.com

MAIL_FROM_ADDRESS=noreply@example.com
MAIL_HOST=smtp.mailtrap.io
MAIL_PORT=2525
MAIL_USERNAME=username
MAIL_PASSWORD=password
MAIL_ENCRYPTION=null

MAILGUN_DOMAIN=
MAILGUN_SECRET=
MAILGUN_ENDPOINT=
POSTMARK_TOKEN=

BROADCAST_DRIVER=log
CACHE_DRIVER=file
QUEUE_CONNECTION=sync
SESSION_DRIVER=file
SESSION_LIFETIME=120
IS_EXTERNAL=false

REDIS_HOST=127.0.0.1
REDIS_PASSWORD=null
REDIS_PORT=6379


REDIS_DB="0"
REDIS_CACHE_DB="1"

ASSET_URL=

TRACKER_SITE_ID=
TRACKER_URL=

APP_NAME=DataImporter

APP_URL=http://localhost

**Reply by iMuhammadMustafa** [reply]
Like I mentioned in the comment above you, I managed to get it solved by using mysql instead of mariadb check my docker compose file.

**Reply by rajviraujla22** [reply]
Now mysql is crashing
![IMG_6009](https://github.com/firefly-iii/firefly-iii/assets/8171496/49719ff2-1fd3-4980-ad0d-85385b5bb048)

**Reply by rajviraujla22** [reply]
Could not poll the database: SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo for db failed: Name or service not known (Connection: mysql, SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1)

This error occured in file /var/www/html/app/Support/FireflyConfig.php on line 65 with code 0.

**Reply by iMuhammadMustafa** [reply]
could you try to clear everything and use my docker compose and change it to match your use (change volumes and environment) and then go on from there?
